src/unlimited.cpp

```diff
@@ -506,7 +508,7 @@ UniValue expedited(const UniValue& params, bool fHelp)
     std::vector<CNode*>::iterator elem = std::find(xpeditedBlkUp.begin(), xpeditedBlkUp.end(),node); 
     if ((flags & EXPEDITED_BLOCKS)&&(flags & EXPEDITED_STOP))
       {
-	if (elem == xpeditedBlkUp.end()) xpeditedBlkUp.erase(elem);
+	if (elem != xpeditedBlkUp.end()) xpeditedBlkUp.erase(elem);
```

Comment:
This is still lacking a test. The fact that the wrong condition could have lived for such a long time is testimony that a test was needed in the first place, and still is.